[#20] Implement user login mechanism by username

Sheptytskyid opened the pull request.

Commit:
integrated sonarqube

zozich committed:
Merge branch 'master' into sonar

Commit:
added some saple code for testing

Commit:
Merge remote-tracking branch 'origin/sonar' into sonar

Commit:
fixed pom

Commit:
try to setup github comments

Commit:
added travis.sh

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
rebuilt according to Travis-ci documentation

Commit:
added sonar to pom

Commit:
fixed travis.yml

Commit:
fixed travis.yml

Commit:
fixed pom

Commit:
added github token

Commit:
added github token

Commit:
put sonar token into travis.yml
removed sonar plugins from pom.xml

Commit:
added sonar-project.properties

Commit:
fixed travis
-sonar-scanner

Commit:
retry

Commit:
retry

Commit:
Merge branch 'sonar'

# Conflicts:
#	src/main/java/com/manardenza/Main.java

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.travis.yml
#	sonar-project.properties

Commit:
[#20] Implement user registration mechanism by username

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -18,6 +19,14 @@ public static HotelDaoImpl getInstance() {
         return instance;
     }
 
+    public ArrayList<Hotel> getHotelsByName(String hotelName) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)

Sheptytskyid replied:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -18,6 +19,14 @@ public static HotelDaoImpl getInstance() {
     }
 
+    public ArrayList<Hotel> getHotelsByName(String hotelName) {
+        return null;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

Sheptytskyid replied:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -18,6 +19,14 @@ public static HotelDaoImpl getInstance() {
+        return null;
+    }
+
+    public ArrayList<Hotel> getHotelsByCity(String hotelCity) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)

Sheptytskyid replied:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -18,6 +19,14 @@ public static HotelDaoImpl getInstance() {
+    }
+
+    public ArrayList<Hotel> getHotelsByCity(String hotelCity) {
+        return null;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

Sheptytskyid replied:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

Sheptytskyid commented:
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 12 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L14](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L14): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L27](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L27): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L28): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L31](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L31): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L32): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L35](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L35): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L39](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L39): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L43](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L43): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/HotelService.java#L44): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserService.java#L10](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2fa2b1000d009dfe336890345e0491e5edc6dce8/src/main/java/com/manardenza/service/UserService.java#L10): Remove this unused "userDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/controller/UserController.java:
```diff
@@ -1,7 +1,7 @@
 package com.manardenza.controller;
 
 import com.manardenza.entity.User;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.manardenza.entity.User'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Commit:
fixed Sonar issues and restored MainTest

Sheptytskyid commented:
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L14](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L14): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L27](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L27): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L28): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L31](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L31): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L32): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L35](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L35): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L39](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L39): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L43](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L43): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/HotelService.java#L44): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [UserService.java#L10](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/2dd3b84c2d2ababd247d0556f936a797eafc79c3/src/main/java/com/manardenza/service/UserService.java#L10): Remove this unused "userDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -18,6 +19,14 @@ public static HotelDaoImpl getInstance() {
     }
 
+    public List<Hotel> getHotelsByName(String hotelName) {
+        return null;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

Sheptytskyid commented on src/main/java/com/manardenza/dao/HotelDaoImpl.java:
```diff
@@ -18,6 +19,14 @@ public static HotelDaoImpl getInstance() {
+    }
+
+    public List<Hotel> getHotelsByCity(String hotelCity) {
+        return new ArrayList<Hotel>();
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)

Commit:
fixed Sonar issues and restored MainTest

Commit:
fixed compil error in UserController

Sheptytskyid commented:
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 12 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L14](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L14): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L27](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L27): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L28): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L31](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L31): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L32): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L35](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L35): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L39](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L39): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L43](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L43): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/4851f21db99452ad873749990684b6367a8a76e4/src/main/java/com/manardenza/service/HotelService.java#L44): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Sheptytskyid commented on src/main/java/com/manardenza/service/UserService.java:
```diff
@@ -0,0 +1,36 @@
+public class UserService {
+
+    private static UserService instance;
+    private UserDaoImpl userDao;
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "userDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Commit:
turn off travis email notifications

Sheptytskyid commented:
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 12 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L14](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L14): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L27](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L27): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L28): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L31](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L31): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L32): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L35](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L35): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L39](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L39): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L43](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L43): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/8799c0c6608abf8e874cb7505f4de186cc0fddd7/src/main/java/com/manardenza/service/HotelService.java#L44): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Commit:
fixed sonar issues

Sheptytskyid commented:
SonarQube analysis reported 10 issues
* ![MAJOR][MAJOR] 10 major

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L14](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L14): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L27](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L27): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L28): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L31](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L31): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L32): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [HotelService.java#L35](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L35): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L39](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L39): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L43](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L43): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/5d9982890a7e1a37bc5e54323bb95b54adaa44eb/src/main/java/com/manardenza/service/HotelService.java#L44): Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

zozich commented on src/main/java/com/manardenza/dao/UserDaoImpl.java:
```diff
@@ -18,8 +19,18 @@ public static UserDaoImpl getInstance() {
     }
 
+    public User getUserByName(String firstName, String lastName) {
+        Optional<User> user = userList.stream().filter(param -> param.getFirstName().equals(firstName) && param
```
This part may be simplified - no need to use if-else, Optional has method .orEsle()

zozich commented on src/main/java/com/manardenza/entity/Room.java:
```diff
@@ -10,8 +10,7 @@
 
-    public Room(long id, String name, int person, int price, Reservation reservation) {
-        super(id);
+    public Room(String name, int person, int price, Reservation reservation) {
```
Room can have many reservations. Fix it here in this PR to be more quick

zozich commented on src/main/java/com/manardenza/service/UserService.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    public void loginUser(String firstName, String lastName) {
+        User user = UserDaoImpl.getInstance().getUserByName(firstName, lastName);
```
Why do you call UserDaoImpl.getInstance()? You already have instance of UserDao stored as instance variable

Commit:
added automatic ID generation for all entities
removed reservation field from Room class
updated DatabaseSeeder, removed IDs
removed MainTest2

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6 into user_registration

# Conflicts:
#	src/main/java/com/manardenza/dao/AbstractDao.java
#	src/main/java/com/manardenza/dao/HotelDaoImpl.java
#	src/main/java/com/manardenza/entity/Room.java
#	src/main/java/com/manardenza/utils/DatabaseSeeder.java

Sheptytskyid commented on src/main/java/com/manardenza/dao/UserDaoImpl.java:
```diff
@@ -3,6 +3,7 @@
 import com.manardenza.entity.User;
 
 import java.util.List;
+import java.util.Optional;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'java.util.Optional'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

Sheptytskyid commented:
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L15](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/19d14336de019ce3b80f7b8fbb3dae76bf3ce48a/src/main/java/com/manardenza/service/HotelService.java#L15): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/19d14336de019ce3b80f7b8fbb3dae76bf3ce48a/src/main/java/com/manardenza/service/HotelService.java#L28): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/19d14336de019ce3b80f7b8fbb3dae76bf3ce48a/src/main/java/com/manardenza/service/HotelService.java#L32): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L36](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/19d14336de019ce3b80f7b8fbb3dae76bf3ce48a/src/main/java/com/manardenza/service/HotelService.java#L36): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L40](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/19d14336de019ce3b80f7b8fbb3dae76bf3ce48a/src/main/java/com/manardenza/service/HotelService.java#L40): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/19d14336de019ce3b80f7b8fbb3dae76bf3ce48a/src/main/java/com/manardenza/service/HotelService.java#L44): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

zozich commented on src/main/java/com/manardenza/service/UserService.java:
```diff
@@ -0,0 +1,34 @@
+        if (user == null) {
+            user = registerNewUser(firstName, lastName);
+        }
+        CurrentUser.getInstance().setUser(user);
```
Store CurrentUser instance in local variable, inject via constructor.

Commit:
removed unused imports

zozich commented on src/main/java/com/manardenza/dao/UserDaoImpl.java:
```diff
@@ -18,6 +19,11 @@ public static UserDaoImpl getInstance() {
     }
 
+    public User getUserByName(String firstName, String lastName) {
+        return database.stream().filter(param -> param.getFirstName().equals(firstName) && param.getLastName()
```
Use additional .filter() statement instead of && - this will make code more readable

Commit:
removed unused imports

Sheptytskyid commented:
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L15](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/22bb5cebcbfbc9a7369525d600fadd3ef3399718/src/main/java/com/manardenza/service/HotelService.java#L15): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/22bb5cebcbfbc9a7369525d600fadd3ef3399718/src/main/java/com/manardenza/service/HotelService.java#L28): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/22bb5cebcbfbc9a7369525d600fadd3ef3399718/src/main/java/com/manardenza/service/HotelService.java#L32): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L36](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/22bb5cebcbfbc9a7369525d600fadd3ef3399718/src/main/java/com/manardenza/service/HotelService.java#L36): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L40](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/22bb5cebcbfbc9a7369525d600fadd3ef3399718/src/main/java/com/manardenza/service/HotelService.java#L40): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/22bb5cebcbfbc9a7369525d600fadd3ef3399718/src/main/java/com/manardenza/service/HotelService.java#L44): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Commit:
fixed getUserByName in UserDaoImpl

zozich reviewed (approved):
Ok now

Sheptytskyid commented:
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [HotelService.java#L15](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/defd4cb3cd541da3ccaee6cf335d369433533289/src/main/java/com/manardenza/service/HotelService.java#L15): Remove this unused "hotelDao" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [HotelService.java#L28](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/defd4cb3cd541da3ccaee6cf335d369433533289/src/main/java/com/manardenza/service/HotelService.java#L28): Remove this unused method parameter "hotelName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L32](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/defd4cb3cd541da3ccaee6cf335d369433533289/src/main/java/com/manardenza/service/HotelService.java#L32): Remove this unused method parameter "cityName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L36](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/defd4cb3cd541da3ccaee6cf335d369433533289/src/main/java/com/manardenza/service/HotelService.java#L36): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L40](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/defd4cb3cd541da3ccaee6cf335d369433533289/src/main/java/com/manardenza/service/HotelService.java#L40): Remove this unused method parameter "roomId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [HotelService.java#L44](https://github.com/Sheptytskyid/JavaCoreFinalProjectGroup6/blob/defd4cb3cd541da3ccaee6cf335d369433533289/src/main/java/com/manardenza/service/HotelService.java#L44): Remove this unused method parameter "params". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
